python/GangaLHCb/Lib/RTHandlers/GaudiExecRTHandlers.py

```diff
@@ -330,10 +331,19 @@ def uploadLocalFile(job, namePattern, localDir, should_del=True):
     """
 
     new_df = DiracFile(namePattern, localDir=localDir)
-    random_SE = random.choice(getConfig('DIRAC')['allDiracSE'])
+    trySEs = getConfig('DIRAC')['allDiracSE']
+    random.shuffle(trySEs)
     new_lfn = os.path.join(getInputFileDir(job), namePattern)
-    returnable = new_df.put(force=True, uploadSE=random_SE, lfn=new_lfn)[0]
-
+    for SE in trySEs:
+        try:
+            returnable = new_df.put(force=True, uploadSE=SE, lfn=new_lfn)[0]
+        except:
+            raise GangaException("Upload of LFN %s to SE %s failed" % (new_lfn, SE)) 
+        if new_df.failureReason == '':
+            break
+        else:
+            logger.warning("Unable to upload LFN %s to SE %s. Trying another SE." % (new_lfn, SE))
+            new_df.failureReason = ''
```

Comment:
Currently the ```put``` doesn't raise an exception if the file upload fails. So the idea is it gets to the end of the list and then the function that called the uploadLocalFile checks the stdout that is returned to it and raises an exception if that doesn't show success.
If for some reason ```put``` does raise an exception then I assume something bad has gone wrong so I raise an exception here. In 6.3 I think this is taken care of rather better.